.github/workflows/release-mvn.yml

```diff
@@ -0,0 +1,36 @@
+name: Release maven package
+
+on:
+  push:
+    branches:
+      - release/*
+
+jobs:
+  pre-release-check:
+    uses: cucumber/.github/.github/workflows/prerelease-checks.yml@main
+
+  build:
+    uses: ./.github/workflows/build.yml
+
+  publish-mvn:
+    name: Publish Maven Package
+    needs: [pre-release-check, build]
+    runs-on: ubuntu-latest
+    environment: Release
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v2
+        with:
+          distribution: "temurin"
+          java-version: "11"
+          cache: "maven"
+
+      - uses: cucumber/action-publish-mvn@v1.0.0
```

Comment:
Yes, it sounds like it. If you can help with that it would be great. I'll set up the repo next week.

Reply:
Sure, as soon as you have the repo I can work on it.

Not sure how to test without actually doing any release but we'll see 😅
I have few Renovate open PRs that can serve as an excuse for doing a patch release anyway.

Reply:
OK I have added the repo and made you an admin.

https://github.com/cucumber/action-publish-sbt

We need to limit the access to these github actions repos as they're a potential vector for supply-chain attacks, so we don't give write access to the commiters team for these repos.

Reply:
Just for my understanding, we're using `sbt` instead of `mvn` for build automation, but are we still publishing to sonatype in the end? I'm just trying to get my head around the pieces of the ecosystem.

Reply:
I also think it's valid to do a patch release just to test the automation.

Reply:
> We need to limit the access to these github actions repos as they're a potential vector for supply-chain attacks, so we don't give write access to the general commiters team for these repos.

Also, assuming we set up CI on the `main` branch to run a test publish, it could be possible to leak those secrets so again we need to lock down who can push to `main` here.

Reply:
Cool, I'll have a look in the coming days.

No issue with limiting access to the action repo, perfectly make sense.

> Just for my understanding, we're using `sbt` instead of `mvn` for build automation, but are we still publishing to sonatype in the end? I'm just trying to get my head around the pieces of the ecosystem.

Absolutely. sbt is very similar to Maven in the sense that it works for the JVM ecosystem and publish to Sonatype. It's juste way more fitted for Scala projects.

For the record, the artifact on Maven central: https://mvnrepository.com/artifact/io.cucumber/cucumber-scala

Reply:
@mattwynne I have a doubt: is the project version supposed to be `x.y.z-SNAPSHOT` before pushing to the branch `release/x.y.z`? Or should the Action retrieve branch name and make a release with the version given in the branch name?

2nd option seems more natural to me but I don't see anything that does something similar in the Maven action or Maven project

Reply:
Good question.

The "modern" process is outlined in https://github.com/cucumber/.github/blob/main/RELEASING.md and there are examples in, say https://github.com/cucumber/tag-expressions/tree/main/.github/workflows

Right now that process is to manually modify the CHANGELOG with the next release number, and manually update the relevant package descriptor, before pushing the commit to a `release/*` branch. So the existing maven action just releases whatever it sees in `pom.xml`.

In fact using the version number in the release branch is just a convention, you could call it `release/foo` and it would still work. Nothing looks at the branch name currently.

The https://github.com/cucumber/action-create-github-release action does read the version number from the CHANGELOG, and potentially we could do the same to get the number needed for the `sbt` release. In fact we have a special GitHub action for that: https://github.com/cucumber/action-get-versions which I was tinkering with for automating more of this, but I hadn't got around to it yet.